src/index.js

```diff
@@ -12,48 +12,41 @@ export default function loader(content) {
 
   const context = options.context || this.rootContext || (this.options && this.options.context);
 
-  let url = loaderUtils.interpolateName(this, options.name, {
+  const originalURL = loaderUtils.interpolateName(this, options.name, {
     context,
     content,
     regExp: options.regExp,
   });
 
-  let outputPath = '';
-
-  if (options.outputPath) {
-    // support functions as outputPath to generate them dynamically
-    outputPath = (
-      typeof options.outputPath === 'function' ? options.outputPath(url) : options.outputPath
-    );
-  }
+  let outputPath = (
+    typeof options.outputPath === 'function' ? options.outputPath(originalURL) : path.join((options.outputPath || '') + originalURL)
+  );
 
   if (options.useRelativePath) {
     const filePath = this.resourcePath;
-    const issuerContext = (this._module && this._module.issuer
-      && this._module.issuer.context) || context;
+    const issuerContext = context || (this._module && this._module.issuer
+      && this._module.issuer.context);
 
     const relativeUrl = issuerContext && path.relative(issuerContext, filePath).split(path.sep).join('/');
 
     const relativePath = relativeUrl && `${path.dirname(relativeUrl)}/`;
     // eslint-disable-next-line no-bitwise
     if (~relativePath.indexOf('../')) {
-      outputPath = path.posix.join(outputPath, relativePath, url);
```

Comment:
Why does this need to be changed ?

Reply:
@michael-ciniawsky because all URL is posix, no need use `posix` here

Reply:
But why was it added then ? Could this break anything ? I'm definitely not against removing it, but I don't get why is eventually needed here in the first place